Reed/Reader/views/ReaderView.swift

```diff
@@ -27,13 +27,24 @@ struct ReaderView: View {
                     data: viewModel.pages,
                     id: \.self,
                     content: { page in
-                        DefinableTextView(
-                            text: .constant(page.content),
-                            tokens: page.tokens,
-                            definerResultHandler: definerResultHandler
-                        )
+                        if viewModel.curPage == -1 || (viewModel.section?.nextNcode != nil && viewModel.curPage == viewModel.pages.endIndex - 1) || (viewModel.section?.prevNcode != nil && viewModel.curPage == 0) {
+                            ProgressView()
+                            .onTapGesture(count: 2) {
+                                doubleTapped()
+                            }
+                        } else {
+                            DefinableTextView(
+                                text: .constant(page.content),
+                                tokens: page.tokens,
+                                definerResultHandler: definerResultHandler
+                            )
+                            .onTapGesture(count: 2) {
+                                doubleTapped()
+                            }
+                        }
                     }
                 )
+                .allowsDragging((viewModel.section?.prevNcode == nil || viewModel.curPage != 0) && (viewModel.section?.nextNcode == nil || viewModel.curPage != viewModel.pages.endIndex - 1))
```

Comment:
Nit : Extra pair of outer parentheses. Line break conditions here, too.

Reply:
fixed conditions, the parentheses is for (cond1 || cond2) && (cond3 || cond4)